src/util/triplea/MapXMLCreator/MapXMLCreator.java

```diff
@@ -0,0 +1,839 @@
+			{
+				if (!MapXMLHelper.s_notes.equals(taNotes.getText()))
+				{
+					MapXMLHelper.s_notes = taNotes.getText().trim();
+				}
+			}
+			
+			@Override
+			public void focusGained(FocusEvent arg0)
+			{
+			}
+		});
+		final JScrollPane spTaNotes = new JScrollPane(taNotes);
+		
+		Dimension size = m_stepActionPanel.getSize();
+		size.height -= 100;
+		spTaNotes.setMinimumSize(size);
+		
+		GridBagConstraints gbc_lFirstRow = new GridBagConstraints();
+		gbc_lFirstRow.insets = new Insets(0, 0, 5, 0);
+		gbc_lFirstRow.gridy = 0;
+		gbc_lFirstRow.gridx = 0;
+		gbc_lFirstRow.anchor = GridBagConstraints.NORTH;
+		gbc_lFirstRow.weightx = 1.0;
+		gbc_lFirstRow.weighty = 1.0;
+		gbc_lFirstRow.gridwidth = 3;
+		m_stepActionPanel.add(spTaNotes, gbc_lFirstRow);
+
+		GridBagConstraints gbc_lSecondRow = (GridBagConstraints) gbc_lFirstRow.clone();
+		gbc_lSecondRow.weighty = 0.0;
+		gbc_lSecondRow.insets = new Insets(0, 0, 0, 0);
+		gbc_lSecondRow.gridwidth = 1;
+		gbc_lSecondRow.gridx = 1;
+		gbc_lSecondRow.gridy = 1;
+		
+		final JButton bPreviewHTML = new JButton("Preview HTML");
+		bPreviewHTML.setPreferredSize(new Dimension(300, 30));
+		bPreviewHTML.setAction(new AbstractAction("Preview HTML")
+		{
+			private static final long serialVersionUID = -5608941822299486808L;
+			
+			public void actionPerformed(final ActionEvent event)
+			{
+				showHTML(MapXMLHelper.s_notes, "HTML Preview");
+			}
+		});
+		m_stepActionPanel.add(bPreviewHTML, gbc_lSecondRow);
+
+		GridBagConstraints gbc_lThirdRow = (GridBagConstraints) gbc_lSecondRow.clone();
+		gbc_lThirdRow.gridy = 2;
+		m_stepActionPanel.add(new JLabel("<html><big>Congratulation!</big></html>"), gbc_lThirdRow);
+		
+		GridBagConstraints gbc_lFourthRow = (GridBagConstraints) gbc_lSecondRow.clone();
+		gbc_lFourthRow.gridy = 3;
+		m_stepActionPanel.add(new JLabel("<html><nobr>You have completed all the steps for creating the map XML.</nobr></html>"), gbc_lFourthRow);
+		
+		final JButton bSave = new JButton("Save Entries to XML");
+		bSave.setPreferredSize(new Dimension(600, 35));
+		bSave.setAction(new AbstractAction("Save Map XML")
+		{
+			private static final long serialVersionUID = -5608941822299486808L;
+			
+			public void actionPerformed(final ActionEvent event)
+			{
+				MapXMLHelper.saveXML();
+			}
+		});
+		GridBagConstraints gbc_lFifthRow = (GridBagConstraints) gbc_lSecondRow.clone();
+		gbc_lFifthRow.insets = new Insets(5, 0, 0, 0);
+		gbc_lFifthRow.gridy = 4;
+		m_stepActionPanel.add(bSave, gbc_lFifthRow);
+	}
+	
+	protected void layoutTabbedPaneWith(final Set<String> keySet)
+	{
+		m_stepActionPanel.setLayout(new BorderLayout());
+		final JTabbedPane tabbedPane = new JTabbedPane(JTabbedPane.SCROLL_TAB_LAYOUT);
+		for (final String key : keySet)
+		{
+			final JPanel innerTabPane = new JPanel();
+			tabbedPane.addTab(key, innerTabPane);
+			ProductionFrontiersPanel.layout(this, innerTabPane, key);
+		}
+		m_stepActionPanel.add(tabbedPane, BorderLayout.CENTER);
+	}
+	
+	
+	/**

```

Comment:
javadoc comment has extra stuff, please clean. IMO you can even skip the citation of where you got the code (unless there is a license restriction otherwise). In the meantime that citation actually makes it hard to change the code.  The first thing a reader needs to do is open a web browser, then verify it is a copy paste, and then verify that the copy paste has not drifted. A person may as well just go through the code and not look at the citation. Thus my recommendation to drop it. 

If there were a lot of extra background information that was needed, the citation could be useful, but the UI code below looks pretty straight forward.